frontend/packages/dev-console/src/components/pipelines/modals/common/validation-utils.ts

```diff
@@ -120,7 +120,24 @@ export const startPipelineSchema = commonPipelineSchema.shape({
       data: volumeTypeSchema,
     }),
   ),
-  secretOpen: yup.boolean().equals([false]),
+  secretForm: yup.object().shape({
+    secretOpen: yup.boolean(),
```

Comment:
Haven't given a good shakedown yet... but the previous state of `secretOpen` was to force the Secret section to be closed before the form was safe to submit. Ie, don't allow them to submit when they were adding a secret (thus not fully adding it). So I'm pretty sure you want to keep it here too unless I am missing something.